app/src/main/kotlin/com/wire/android/feature/auth/login/LoginViewModel.kt

```diff
@@ -16,20 +21,24 @@ import com.wire.kalium.logic.feature.auth.LoginUseCase
 @HiltViewModel
 class LoginViewModel @Inject constructor(
     private val loginUseCase: LoginUseCase,
-    private val savedStateHandle: SavedStateHandle
+    private val savedStateHandle: SavedStateHandle,
+    private val navigationManager: NavigationManager
 ) : ViewModel() {
 
-    private val userIdentifier = mutableStateOf(
+    var userIdentifier by mutableStateOf(
         savedStateHandle.get(USER_IDENTIFIER_SAVED_STATE_KEY) ?: String.EMPTY
     )
-    private val password = mutableStateOf(String.EMPTY)
+        private set
+
+    var password by mutableStateOf(String.EMPTY)
+        private set
```

Comment:
you could create a State object as a plain Kotlin Class and wrap it into MutableStateFlow()

data LoginState( val password : String, val userIdentifier : String) and consume it inside the LoginContent for example with collectAsState method()

Reply:
Do we have to mix coroutine `Flow` and compose `State` just to keep the values in the ViewModel?

Reply:
I guess depends on a use case, but now you say it is not really required here I tihnk but MutableStateFlow gives you more option to use operators as combine and stuff.

Reply:
i don't see it as necessary since `user Identifier` and `password` are storing the state of a `TextField`, which start acting weird (the text was duplicated after each input) when assigning a string from viewModel as value e.g. 
```kotlin
@Composable
private fun UserIdentifierInput(modifier: Modifier, userIdentifier: String, onValueChange: (TextFieldValue) -> Unit) {
    WireTextField(
        value = TextFieldValue(userIdentifier), 
        ...
    )
}

use:
UserIdentifierInput(
     modifier = Modifier.fillMaxWidth(),
      userIdentifier = loginViewModel.userIdentifier, // loginViewModel.userIdentifier in this case is a string  mutableState 

```